scripts/pylib/twister/twisterlib/testinstance.py

```diff
@@ -3,19 +3,27 @@
 # Copyright (c) 2018-2022 Intel Corporation
 # Copyright 2022 NXP
 # SPDX-License-Identifier: Apache-2.0
-
+from __future__ import annotations
```

Comment:
suggest to change the annotation changes to another PR

Reply:
Here I wanted only to add annotations to improve code completion and readability in places, I changed. Just to do a small step, do not add annotations in the rest of the code. If using annotations in that PR is a blocker, I will remove it and add it in separate PR (or not, if there is no plan to use annotation in twister to not mess the coding style)

Reply:
@gchwier add annotation is good, but maybe in a more consistent way. @PerMac  what do you think?

Reply:
@PerMac I mean do we need to create a general task to add anotation to all twisters apis?

Reply:
> I heard several times an advice from seniors that it is good to do such "refactoring" iteratively.

This should be set as a goal first and there needs to be an agreement that we are going this way in general, it should not be left up to individual preferences and choices. 

> Refactoring everything in one go can be a long and potentially error prone proces.

And doing this in an unmanaged way is confusing and makes code very difficult to read and maintain.

> I also saw similar approach already done for twister [#58046 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/58046#discussion_r1202238875) It wasn't annotation related but in general few changes resolving IDE complains.

Not sure how this is related. This seems to be a bug fix in this case.

I am fine with this overall, but we should avoid having different styles in the code and we need to agree on the goals and figure out a roadmap to achieve them.